extensions/sql-migration/src/models/stateMachine.ts

```diff
@@ -961,65 +954,204 @@ export class MigrationStateModel implements Model, vscode.Disposable {
+					return <azureResource.AzureResourceResourceGroup>{
+						id: getFullResourceGroupFromId(vm.id),
+						name: getResourceGroupFromId(vm.id),
+						subscription: {
+							id: vm.subscriptionId
+						},
+						tenant: vm.tenantId,
+					};
+				}));
+
+				// remove duplicates
+				this._resourceGroups = this._resourceGroups.filter((v, i, a) => a.findIndex(v2 => (v2.id === v.id)) === i);
+				sortResourceArrayByName(this._resourceGroups);
+			} else {
+				this._resourceGroups = [];
+			}
+
+			this._resourceGroups.forEach((rg) => {
+				resourceGroupValues.push({
+					name: rg.id,
+					displayName: rg.name
+				});
+			});
+
+			if (resourceGroupValues.length === 0) {
+				resourceGroupValues = [
+					{
+						displayName: constants.RESOURCE_GROUP_NOT_FOUND,
+						name: ''
+					}
+				];
+			}
+		} catch (e) {
+			console.log(e);
+			resourceGroupValues = [
+				{
+					displayName: constants.RESOURCE_GROUP_NOT_FOUND,
+					name: ''
+				}
+			];
+		}
+		return resourceGroupValues;
+	}
+
+	public async getAzureResourceGroupForStorageAccountsDropdownValues(subscription: azureResource.AzureResourceSubscription): Promise<azdata.CategoryValue[]> {
+		let resourceGroupValues: azdata.CategoryValue[] = [];
+		try {
+			if (this._azureAccount && subscription) {
+				let storageAccounts = await getAvailableStorageAccounts(this._azureAccount, subscription);
+				this._resourceGroups = await Promise.all(storageAccounts.map(async (sa) => {
+					return <azureResource.AzureResourceResourceGroup>{
+						id: getFullResourceGroupFromId(sa.id),
+						name: getResourceGroupFromId(sa.id),
+						subscription: {
+							id: sa.subscriptionId
+						},
+						tenant: sa.tenantId,
+					};
+				}));
+
+				// remove duplicates
+				this._resourceGroups = this._resourceGroups.filter((v, i, a) => a.findIndex(v2 => (v2.id === v.id)) === i);
+				sortResourceArrayByName(this._resourceGroups);
+			} else {
+				this._resourceGroups = [];
+			}
+
+			this._resourceGroups.forEach((rg) => {
+				resourceGroupValues.push({
+					name: rg.id,
+					displayName: rg.name
+				});
+			});
+
+			if (resourceGroupValues.length === 0) {
+				resourceGroupValues = [
+					{
+						displayName: constants.RESOURCE_GROUP_NOT_FOUND,
+						name: ''
+					}
+				];
+			}
+		} catch (e) {
+			console.log(e);
+			resourceGroupValues = [
+				{
+					displayName: constants.RESOURCE_GROUP_NOT_FOUND,
+					name: ''
 				}
+			];
+		}
+		return resourceGroupValues;
+	}
+
+	public async getAzureResourceGroupForSqlMigrationServicesDropdownValues(subscription: azureResource.AzureResourceSubscription): Promise<azdata.CategoryValue[]> {
+		let resourceGroupValues: azdata.CategoryValue[] = [];
+		try {
+			if (this._azureAccount && subscription) {
+				let dmsInstances = await getSqlMigrationServices(this._azureAccount, subscription);
+				this._resourceGroups = await Promise.all(dmsInstances.map(async (dms) => {
```

Comment:
same issue here, I don't see an async call inside the .map callback

Reply:
Fixed, removed all the unnecessary async calls